### PR TITLE
Service discovery - SchedulingProperties

### PR DESCRIPTION
**Pull request by eishay**
Allow scheduling tasks using ZK's leadership property

### CODE REVIEW

File: shoebox/app/com/keepit/common/plugin/SchedulingPlugin.scala [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+  private def sendMessage(receiver: ActorRef, message: Any): Unit = if (schedulingProperties.allowSchecualing) {
+    log.info(s"sending a scheduled message $message to actor")
+    receiver ! message
+  }
+
   def scheduleTask(system: ActorSystem, initialDelay: FiniteDuration,
       frequency: FiniteDuration, receiver: ActorRef, message: Any) {
-    if (schedulerIsEnabled)
-      _cancellables :+= system.scheduler.schedule(initialDelay, frequency, receiver, message)
+    if (schedulingProperties.neverAllowSchecualing) return //don't even schedule in this case
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/plugin/SchedulingPlugin.scala]: stylistically, I prefer not to have returns in the middle like this unless it really makes the code simpler

File: shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package com.keepit.common.plugin
+
+import com.keepit.common.zookeeper._
+import com.google.inject.{Inject, Singleton, ImplementedBy}
+
+@ImplementedBy(classOf[SchedulingPropertiesImpl])
+trait SchedulingProperties  {
+  def allowSchecualing: Boolean
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala]: Is this just a creative spelling of `allowScheduling`?

File: shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  def neverAllowSchecualing: Boolean = allowSchecualing
+}
+
+class SchedulingPropertiesImpl @Inject() (
```
Comment (gmethvin) [inline shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala:13]: should this be `@Singleton`?

**Comment by gmethvin** [comment]
looks good to me

File: shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala (hunk 2 above)
Comment (eishay) [inline shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala]: :-)

File: shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala (hunk 3 above)
Comment (eishay) [inline shoebox/app/com/keepit/common/plugin/SchedulingProperties.scala:13]: sure

**Pull request merged**